### PR TITLE
Add stale GH Action for marking issues and closing PRs unless assigned

### PR DESCRIPTION
**Pull request by encima**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

[Stale](https://github.com/actions/stale) bot added to actions workflow to run on a schedule 

## What is the current behavior?

No automatic activity for issues or PRs

## What is the new behavior?

- PRs are closed if:
	- They do **not**  have the label: `dnc`
	- They are **not** assigned
	- They are inactive for more than 30 days
- Issues are labelled as `stale` if:
	- They do **not** have the label: `dnc` or `needs-analysis`
	- They are inactive for more than 30 days
	- They as **not** assigned 
- Issues are closed if:
	- They are labelled as `stale`
	- They remain inactive for 10 days after being labelled as `stale`

### CODE REVIEW

**Review by pamelachia — APPROVED** [review]

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Close Stale Issues
+        uses: actions/stale@v9.0.0
+        with:
+          days-before-pr-close: 30
+          days-before-issue-stale: 30
+          days-before-issue-close: 10
+          stale-issue-message: 'After 30 days of inactivity, this issue has been marked as stale. This issue will be auto-closed in 10 days. Comment, assign or remove the stale label to prevent this.'
```
Comment (kamilogorek) [inline .github/workflows/stale.yml:16]: `stale` label is automatically assigned after 30 days, so saying `assign or remove the stale label` sounds off

Comment (encima) [reply .github/workflows/stale.yml:16]: Ah true, terminology is tough! Assign (in my mind) meant assign the PR to a person and not adding a label. All changes welcome!

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          stale-issue-label: stale
+          close-issue-label: staled
+          close-pr-label: staled
```
Comment (kamilogorek) [inline .github/workflows/stale.yml:19]:
```suggestion
          days-before-pr-stale: 30
          days-before-pr-close: 7
          days-before-issue-stale: 30
          days-before-issue-close: 7
          stale-pr-message: 'This pull request has been inactive for 30 days and will be closed in another week. If you want to give it another month, leave a comment, assign it to someone, or remove a `Stale` label.'
          stale-issue-message: 'This issue has been inactive for 30 days and will be closed in another week. If you want to give it another month, leave a comment, assign it to someone, or remove a `Stale` label.'
```

Comment (kamilogorek) [reply .github/workflows/stale.yml:19]: - added `days-before-pr-stale` to make the behavior in-line with issues
- changed close to `7` days as it better aligns with people's behavior; its easier to read it as "ok, I have another week" than "hmm, when 10 days land exactly?"
- removed `close` labels, as its already possible to query for issues/prs that are closed+labeled with `Stale`
- removed `stale` label overrides, as it's already a default

**Pull request closed without merging**